Assets/BossRoom/Scripts/Server/Game/AIState/AttackAIState.cs

```diff
@@ -122,9 +134,27 @@ private ServerCharacter ChooseFoe()
             return closestFoe;
         }
 
+        /// <summary>
+        /// Chooses a usable attack and returns an ActionDescription representing it. If no actions are usable, returns null.
+        /// </summary>
+        /// <returns>Action to use, or null</returns>
         private ActionDescription GetCurrentAttackInfo()
         {
-            m_CurAttackAction = m_AttackActions[Random.Range(0, m_AttackActions.Count)];
+            m_TempListOfAttackActions.Clear();
```

Comment:
The only thing that I don't like here is the fact that Getter's (in my opinion) should not modify the members of a class, it just violates multiple principles. I propose to rename this function.

Reply:
Renamed `GetCurrentAttackInfo()` to `ChooseAttack()` and cleaned up some leftover code. (The function can now just return an enum value instead of needing to set a member variable.)